nx/lib/nx/shared.ex

```diff
@@ -329,6 +329,13 @@ defmodule Nx.Shared do
   def binary_type(a, b) when is_number(b), do: Nx.Type.merge_scalar(type(a), b)
   def binary_type(a, b), do: Nx.Type.merge(type(a), type(b))
 
+  def binary_type(_a, _b, :power), do: {:f, 64}
+
+  def binary_type(a, _b, op) when op in [:bitwise_and, :bitwise_or, :bitwise_xor],
+    do: type(a)
```

Comment:
Instead of changing it here, change the code in `Nx` so bitwise_and, bitwise_or, and bitwise_xor call a new function called `element_wise_bit_op` that has the new type logic. We will also need a new `Nx.Type.merge_bitwise/2` function?